test/Billing.Test/Controllers/FreshsalesControllerTests.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Bit.Billing.Controllers;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories;
+using Bit.Core.Settings;
+using Bit.Test.Common.AutoFixture.Attributes;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Billing.Test.Controllers
+{
+    public class FreshsalesControllerTests
+    {
+        private const string ApiKey = "TEST_FRESHSALES_APIKEY";
+        private const string TestLead = "TEST_FRESHSALES_TESTLEAD";
+
+        private static (FreshsalesController sut, IUserRepository userRepository, IOrganizationRepository organizationRepository) CreateSut(
+            string freshsalesApiKey)
+        {
+            var userRepository = Substitute.For<IUserRepository>();
+            var organizationRepository = Substitute.For<IOrganizationRepository>();
+
+
+            var billingSettings = Options.Create(new BillingSettings
+            {
+                FreshsalesApiKey = freshsalesApiKey,
+            });
+            var globalSettings = new GlobalSettings();
+            globalSettings.BaseServiceUri.Admin = "https://test.com";
+
+            var sut = new FreshsalesController(
+                userRepository,
+                organizationRepository,
+                billingSettings,
+                Substitute.For<ILogger<FreshsalesController>>(),
+                globalSettings
+            );
+
+            return (sut, userRepository, organizationRepository);
+        }
+
+        [RequiredEnvironmentTheory(ApiKey, TestLead), EnvironmentData(ApiKey, TestLead)]
```

Comment:
I hadn't realized there was a `Skip` property on `DataAttribute` that I could use but I tried tapping into that and simplifying it so that it would only need one of these but I couldn't get the `Skip` property to be respected. This issue xunit/visualstudio.xunit#266 makes it seem like it's a known issue with Visual Studio until 2.4.2 and we are on 2.4.1 but I am testing it from `dotnet test` and and VS for Mac. So I am out of ideas on how to combine these two. Any ideas?

Reply:
well that's disappointing. I didn't realize that attribute doesn't work. I think what you have here is the best workaround available.